test cases/common/228 link with custom target/meson.build

```diff
@@ -0,0 +1,7 @@
+project('link with custom targets', 'c')
+t = custom_target('dummy', output: [ 'libdummy.a' ], command: [ 'echo', 'dummy' ])
+dep1 = declare_dependency(link_with: t)
+dep2 = declare_dependency(link_with: t[0])
+
+static_library('lib1', 'empty.c', dependencies: dep1, install: true)
+static_library('lib2', 'empty.c', dependencies: dep2, install: true)
```

Comment:
Can you please add this into `215 link custom`, grouping tests helps reduce CI run time.

Reply:
See 7dd302773dd2d6443ab52d608d1c86aef578f280 where i fixed/tested a similar case.

Reply:
This still needs to be addressed @Sahnvour